src/components/Dropdown.tsx

```diff
@@ -0,0 +1,157 @@
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
+/* eslint-disable unicorn/no-null */
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable react/no-unused-prop-types */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+
+import type { MouseEventHandler } from 'react';
+import { useState } from 'react';
+import type {
+  GroupBase,
+  OnChangeValue,
+  OptionsOrGroups,
+  PropsValue
+} from 'react-select';
+import Select, { createFilter } from 'react-select';
+import { Icon } from './Icon';
+import { Label } from './Label';
+
+interface SelectOption {
+  value: number | string;
+  label: string;
+  onSelect?: () => void;
+}
+
+/**
+ * Utility Functions
+ */
+
+function onCloser(
+  index: number,
+  onChange: (result: PropsValue<SelectOption>) => void,
+  selected?: PropsValue<SelectOption>
+): MouseEventHandler<HTMLLIElement> {
+  return () => {
+    if (!selected || !Array.isArray(selected)) return;
+    const result = selected.filter((_, index_) => index_ !== index);
+    onChange(result);
+  };
+}
+
+const filterOptions = (
+  options: PropsValue<SelectOption>,
+  selected: PropsValue<SelectOption>,
+  isMulti: boolean
+): OptionsOrGroups<SelectOption, GroupBase<SelectOption>> => {
+  if (!selected || !options)
+    return options as OptionsOrGroups<SelectOption, GroupBase<SelectOption>>;
+
+  if (isMulti)
+    return (options as SelectOption[]).filter(
+      o => !(selected as SelectOption[]).map(s => s.value).includes(o.value)
+    );
+
+  return options as OptionsOrGroups<SelectOption, GroupBase<SelectOption>>;
+};
+
+/**
+ * Supporting Components
+ */
+
+interface PillProperties {
+  value: string;
+  onClose: MouseEventHandler<HTMLLIElement>;
+}
+const Pill = ({ value, onClose }: PillProperties): JSX.Element => (
+  <li className='pill' onClick={onClose}>
+    <Label htmlFor={value} inline>
+      {value}
+      <div>
+        <Icon name='error' />
+      </div>
+    </Label>
+  </li>
+);
+
+interface PillsProperties {
+  selected: PropsValue<SelectOption>;
+  isMulti: boolean;
+  onChange: (event: PropsValue<SelectOption>) => void;
+}
+const Pills = ({
+  selected,
+  isMulti,
+  onChange
+}: PillsProperties): JSX.Element | null => {
+  if (
+    !isMulti ||
+    !selected ||
+    (Array.isArray(selected) && selected.length === 0)
+  )
+    return null;
+  return (
+    <div className='o-multiselect_choices pills'>
```

Comment:
This should prob be a `<ul>` instead of a `<div>` to play nicer with its children `<li>`.

Reply:
Updated.